Userland/Utilities/stty.cpp

```diff
@@ -0,0 +1,627 @@
+            t.c_cc[VKILL] = CKILL;
+        } else if (parameters[parameter_idx] == "sane") {
+            t.c_iflag = TTYDEF_IFLAG;
+            t.c_oflag = TTYDEF_OFLAG;
+            t.c_cflag = TTYDEF_CFLAG;
+            t.c_lflag = TTYDEF_LFLAG;
+            for (size_t i = 0; i < NCCS; ++i)
+                t.c_cc[i] = ttydefchars[i];
+            t.c_ispeed = t.c_ospeed = TTYDEF_SPEED;
+        } else {
+            auto maybe_error = parse_flag_or_char();
+            if (maybe_error.is_error())
+                return maybe_error.error();
+        }
+        ++parameter_idx;
+    }
+    return {};
+}
+
+int main(int argc, char** argv)
+{
+    if (pledge("stdio tty rpath", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    if (unveil("/dev", "r") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    if (unveil(nullptr, nullptr) < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    String device_file;
+    bool stty_readable = false;
+    bool all_settings = false;
+
+    // Core::ArgsParser can't handle the weird syntax of stty, so we use getopt_long instead.
+    opterr = 0; // We handle unknown flags gracefully by starting to parse the arguments in `apply_modes`.
+    int optc;
+    bool should_quit = false;
+    while (!should_quit && ((optc = getopt_long(argc, argv, "-agF:", long_options, nullptr)) != -1)) {
+        switch (optc) {
+        case 'a':
+            all_settings = true;
+            break;
+        case 'g':
+            stty_readable = true;
+            break;
+        case 'F':
+            if (!device_file.is_empty()) {
+                warnln("Only one device may be specified");
+                exit(1);
+            }
+            device_file = optarg;
+            break;
+        default:
+            should_quit = true;
+            break;
+        }
+    }
+
+    if (stty_readable && all_settings) {
+        warnln("Save mode and all-settings mode are mutually exclusive");
+        exit(1);
+    }
+
+    int terminal_fd = STDIN_FILENO;
+    if (!device_file.is_empty()) {
+        if ((terminal_fd = open(device_file.characters(), O_RDONLY, 0)) < 0) {
+            perror("open");
+            exit(1);
+        }
+    }
+
+    ScopeGuard file_close_guard = [&] { close(terminal_fd); };
+
+    termios initial_termios;
+    if (tcgetattr(terminal_fd, &initial_termios) < 0) {
+        perror("tcgetattr");
+        exit(1);
+    }
+
+    winsize initial_winsize;
+    if (ioctl(terminal_fd, TIOCGWINSZ, &initial_winsize) < 0) {
+        perror("ioctl(TIOCGWINSZ)");
+        exit(1);
+    }
+
+    if (optind < argc) {
+        if (stty_readable || all_settings) {
+            warnln("Modes cannot be set when printing settings");
+            exit(1);
+        }
+        auto nothing_or_error = apply_modes(argc - optind, argv + optind, initial_termios, initial_winsize);
+        if (nothing_or_error.is_error())
+            return nothing_or_error.error();
```

Comment:
Nitpick: `nothing_or_error` sounds weird, how about this:
```suggestion
        if (auto result = apply_modes(argc - optind, argv + optind, initial_termios, initial_winsize); result.is_error())
            return result.error();
```

Reply:
Fixed it, thank you for the review!